critical_demand.py

```diff
@@ -35,513 +35,892 @@
-from oemof_visio import ESGraphRenderer
+    df_results = df_costs.copy()
+    df_results["capacity"] = 0
+    df_results["total_flow"] = 0
+    df_results["cash_flow"] = 0
+    df_results["cash_flow"] = 0
 
-es = ESGraphRenderer(energy_system, legend=True, filepath=f"case_{case}.pdf")
-es.render()
+    project_lifetime = 20
+    wacc = 0.06
+    CRF = annuity(1, project_lifetime, wacc)
 
-model = solph.Model(energy_system)
-model.solve(
-    solver=solver,
-    solve_kwargs={"tee": True},
-    cmdline_options=solver_option[solver],
-)
+    results_pv = solph.views.node(results=results, node="pv")
+    if case in (case_D, case_DBPV):
+        results_diesel_source = solph.views.node(results=results, node="diesel_source")
+        results_diesel_genset = solph.views.node(results=results, node="diesel_genset")
 
-# End of the calculation time.
-end_simulation_time = time.time()
+    results_inverter = solph.views.node(results=results, node="inverter")
+    results_rectifier = solph.views.node(results=results, node="rectifier")
+    if case in (case_BPV, case_DBPV):
+        results_battery = solph.views.node(results=results, node="battery")
 
+    results_demand_el = solph.views.node(results=results, node="electricity_demand")
+    results_critical_demand_el = solph.views.node(
+        results=results, node="electricity_critical_demand"
+    )
+    results_excess_el = solph.views.node(results=results, node="excess_el")
+
+    # -------------------- SEQUENCES (DYNAMIC) --------------------
+    # Hourly demand profile.
+    sequences_demand = results_demand_el["sequences"][
+        (("electricity_ac", "electricity_demand"), "flow")
+    ]
+
+    sequences_critical_demand = results_critical_demand_el["sequences"][
+        (("electricity_ac", "electricity_critical_demand"), "flow")
+    ]
+
+    if case in (case_BPV, case_DBPV):
+        # Hourly profiles for solar potential and pv production.
+        sequences_pv = results_pv["sequences"][(("pv", "electricity_dc"), "flow")]
+        df_results.loc["pv", "total_flow"] = sequences_pv.sum()
+
+        # TODO find what we would like to have here
+        df_results.loc["battery", "total_flow"] = 0
+
+    if case in (case_D, case_DBPV):
+        # Hourly profiles for diesel consumption and electricity production
+        # in the diesel genset.
+        # The 'flow' from oemof is in kWh and must be converted to
+        # kg by dividing it by the lower heating value and then to
+        # liter by dividing it by the diesel density.#ADN:??
+        sequences_diesel_consumption = (
+            results_diesel_source["sequences"][(("diesel_source", "diesel"), "flow")]
+            / diesel_lhv
+            / diesel_density
+        )
 
-##########################################################################
-# Process the results
-##########################################################################
+        df_results.loc["diesel_genset", "cash_flow"] = (
```

Comment:
@adnanalakori This is the definition of cash_flow for diesel, you can look in the file where "cash_flow" is used (using ctrl+f function to search for occurences)